tests/grouping/test_group_observables.py

```diff
@@ -301,3 +303,93 @@ def test_binary_repr_custom_wire_map(self):
         _ = grouping_instance.binary_repr(n_qubits, wire_map)
 
         assert grouping_instance._wire_map == wire_map
+
+    def test_return_list_coefficients(self):
+        """Tests that if the coefficients are given as a list, the groups
+        are likewise lists."""
+        obs = [qml.PauliX(0), qml.PauliX(1)]
+        coeffs = [1., 2.]
+        _, grouped_coeffs = group_observables(obs, coeffs)
+        assert isinstance(grouped_coeffs[0], list)
+
+
+class TestDifferentiable:
+    """Tests that grouping observables is differentiable with respect to the coefficients."""
+
+    def test_differentiation_autograd(self, tol):
+        """Test that grouping is differentiable with autograd tensors as coefficient"""
+        coeffs = pnp.array([1.0, 2.0, 3.0], requires_grad=True)
+        obs = [PauliX(wires=0), PauliX(wires=1), PauliZ(wires=1)]
+
+        def group(coeffs, select=None):
+            _, grouped_coeffs = qml.grouping.group_observables(obs, coeffs)
+            return grouped_coeffs[select]
+
+        jac_fn = qml.jacobian(group)
+        assert pnp.allclose(
+            jac_fn(coeffs, select=0), pnp.array([[1.0, 0.0, 0.0], [0.0, 1.0, 0.0]]), atol=tol
+        )
+        assert pnp.allclose(jac_fn(coeffs, select=1), pnp.array([[0.0, 0.0, 1.0]]), atol=tol)
```

Comment:
Out of curiosity, would the output of `group` simply be `permutation matrix @ original coeffs`? So computing the jacobian simply returns the permutation matrix?

Reply:
I guess so! Just with a chopped up shape...